tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/PrettyWriter.scala

```diff
@@ -32,24 +33,38 @@ class PrettyWriter(writer: PrintWriter, textWidth: Int = 80, indent: Int = 2) ex
   // the stack of lambda declarations
   private var lambdaStack: List[TlaOperDecl] = Nil
 
-  def write(mod: TlaModule): Unit = {
-    writer.write(pretty(toDoc(mod), textWidth).layout)
-  }
+  def prettyWriteDoc(doc: Doc): Unit = writer.write(pretty(doc, textWidth).layout)
 
-  def write(decl: TlaOperDecl): Unit = {
-    writer.write(pretty(toDoc(decl), textWidth).layout)
-  }
+  def write(string: String): Unit = prettyWriteDoc(text(string))
```

Comment:
This is a way to mess with the pretty printer.

Reply:
Perhaps, but also the only way to force comments in at the desired line, since comments are invisible in the internal representation.

Reply:
I believe that comments deserve a method of their own. No big deal if they are printed on a new line. At least, they will not ruin pretty printing.